src/styles/getTooltipStyles.js

```diff
@@ -106,47 +105,53 @@ export default function getTooltipStyles(
-    const horizontalShift = computeHorizontalShift(windowWidth, xOffset, componentWidth, tooltipWidth, manualShiftHorizontal);
-
-    // Determine if we need to shift the pointer horizontally to prevent it from being too near to the edge of the tooltip
-    // We shift it to the right a bit if the tooltip is positioned on the extreme left
-    // and shift it to left a bit if the tooltip is positioned on the extreme right.
-    const horizontalShiftPointer =
-        horizontalShift > 0
-            ? Math.max(-horizontalShift, -(tooltipWidth / 2) + POINTER_WIDTH / 2 + variables.componentBorderRadiusSmall)
-            : Math.min(-horizontalShift, tooltipWidth / 2 - POINTER_WIDTH / 2 - variables.componentBorderRadiusSmall);
-
     const tooltipVerticalPadding = spacing.pv1;
-    const tooltipFontSize = variables.fontSizeSmall;
 
-    // We get wrapper width based on the tooltip's inner text width so the wrapper is just big enough to fit text and prevent white space.
-    // If the text width is less than the maximum available width, add horizontal padding.
-    // Note: tooltipContentWidth ignores the fractions (OffsetWidth) so add 1px to fit the text properly.
-    const wrapperWidth = tooltipContentWidth && tooltipContentWidth + spacing.ph2.paddingHorizontal * 2 + 1;
+    // We get tooltip width and height based on the tooltip's content width and height
+    // so the tooltip wrapper is just big enough to fit content and prevent white space.
+    const tooltipWidth = tooltipContentWidth && tooltipContentWidth + spacing.ph2.paddingHorizontal * 2;
+    const tooltipHeight = tooltipContentHeight && tooltipContentHeight + tooltipVerticalPadding.paddingVertical * 2;
 
     // Hide the tooltip entirely if it's position hasn't finished measuring yet. This prevents UI jank where the tooltip flashes in the top left corner of the screen.
     const opacity = xOffset === 0 && yOffset === 0 ? 0 : 1;
 
-    const isTooltipSizeReady = tooltipWidth !== 0 && tooltipHeight !== 0;
-    const scale = !isTooltipSizeReady ? 1 : currentSize;
+    const isTooltipSizeReady = tooltipWidth !== undefined && tooltipHeight !== undefined;
+    let shouldShowBelow = false;
+    let scale = 1;
+    let horizontalShift = 0;
+    let horizontalShiftPointer = 0;
     let wrapperTop = 0;
     let wrapperLeft = 0;
+    let pointerWrapperTop = 0;
+    let pointerWrapperLeft = 0;
+    let pointerAdditionalStyle = {};
 
     if (isTooltipSizeReady) {
+        // Determine if the tooltip should display below the wrapped component.
+        // If either a tooltip will try to render within GUTTER_WIDTH logical pixels of the top of the screen,
+        // Or the wrapped component is overlapping at top-left with another element
+        // we'll display it beneath its wrapped component rather than above it as usual.
+        shouldShowBelow = yOffset - tooltipHeight < GUTTER_WIDTH || isOverlappingAtTop(xOffset, yOffset);
+
+        scale = currentSize;
```

Comment:
Can you please add a comment for _why_ the scale is changed to the currentSize here (and not when it's initialized).

Reply:
Added.